src/test/java/com/skillstorm/spartanwireless/services/impl/TestCustomerServiceImpl.java

```diff
@@ -1,56 +1,85 @@
-        // TODO Create a customers list
-        cs = new CustomerServiceImpl(new TestCustomerRepo());
-    }
-    
-    @Test
-    public void TestCreateCustomer() {
-        cs.createCustomer(null); // TODO Implement createCustomer Test.
-        assertEquals(getClass(), getClass());
+        custId1 = 1L;
+        customer1 = Customer.builder().custId(1L).name("Bob").address("121 Rock Rd.").email("bob@gmail.com").isArchived(false).build();
+        customer2 = Customer.builder().custId(2L).name("Gil").address("59 Scenic St.").email("gil@gmail.com").isArchived(false).build();
+        customerRequestDto1 = CustomerRequestDto.builder().name("Bob").address("121 Rock Rd.").email("bob@gmail.com").build();
     }
 
     @Test
-    public void TestGetAllCustomers() {
-        // TODO Create a test for Service's getAllCustomers method
+    public void CustomerService_CreateCustomer_ReturnsCustomerResponseDto() {
+        when(customerRepository.save(any(Customer.class))).thenReturn(customer1);
+        CustomerResponseDto customerResponseDto = customerServiceImpl.createCustomer(customerRequestDto1);
+        assertNotNull(customerResponseDto);
     }
 
     @Test
-    public void TestGetAllActiveCustomers() {
-        // TODO Create a test for Service's getAllActiveCustomers method
+    public void CustomerService_GetAllActiveCustomers_ReturnsListOfCustomerReponseDto() {
+        List<Customer> customers = new ArrayList<>();
+        customers.add(customer1);
+        customers.add(customer2);
+        when(customerRepository.findAll()).thenReturn(customers);
+        List<CustomerResponseDto> customerResponseDtos = customerServiceImpl.getAllActiveCustomers();
+        assertNotNull(customerResponseDtos);
     }
 
     @Test
-    public void TestGetCustomerById() {
-        // TODO Create test for Service's getCustomerById method
+    public void CustomerService_FindById_ReturnCustomerResponseDto() {
+        when(customerRepository.findById(custId1)).thenReturn(Optional.ofNullable(customer1));
+        CustomerResponseDto customerResponseDto = customerServiceImpl.getCustomerById(custId1);
+        assertNotNull(customerResponseDto);
     }
-    
+
     @Test
-    public void TestUpdateCustomer() {
-        // TODO Create test for Repo's updateCustomer method
+    public void CustomerService_UpdateCustomer_ReturnCustomerResponseDto() {
+        when(customerRepository.save(any(Customer.class))).thenReturn(customer1);
+        CustomerResponseDto customerResponseDto = customerServiceImpl.updateCustomer(custId1, customerRequestDto1);
+        assertNotNull(customerResponseDto);
     }
 
     @Test
-    public void TestDeleteCustomerById() {
-        // TODO Create test for Service's deleteCustomerById method
-        // Remember to use isArchived functionality
+    public void CustomerService_DeleteCustomer_ReturnVoid() {
+        customerServiceImpl.deleteCustomerById(custId1);
+        verify(customerRepository, times(1)).deleteById(custId1);
```

Comment:
Miss this last time. We need to verify that it's not deleting, but archiving.

Reply:
yeah i need to implement that logic.